evap/staff/tests/test_importers.py

```diff
@@ -125,43 +126,73 @@ def test_import_makes_inactive_user_active(self):
             excel_content = excel_file.read()
 
-        success_messages, warnings, errors = EnrollmentImporter.process(excel_content, semester, None, None, test_run=True)
+        success_messages, warnings, errors = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=True)
         self.assertIn("The import run will create 23 courses/evaluations and 23 users:", "".join(success_messages))
         # check for one random user instead of for all 23
         self.assertIn("Ferdi Itaque (ferdi.itaque)", "".join(success_messages))
         self.assertEqual(errors, [])
         self.assertEqual(warnings, {})
 
-        success_messages, warnings, errors = EnrollmentImporter.process(excel_content, semester, vote_start_datetime, vote_end_date, test_run=False)
+        old_user_count = UserProfile.objects.all().count()
+
+        success_messages, warnings, errors = EnrollmentImporter.process(excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False)
         self.assertIn("Successfully created 23 courses/evaluations, 6 students and 17 contributors:", "".join(success_messages))
         self.assertIn("Ferdi Itaque (ferdi.itaque)", "".join(success_messages))
         self.assertEqual(errors, [])
         self.assertEqual(warnings, {})
 
+        self.assertEqual(Evaluation.objects.all().count(), 23)
+        expected_user_count = old_user_count + 23
+        self.assertEqual(UserProfile.objects.all().count(), expected_user_count)
+
+    def test_degrees_are_merged(self):
+        with open(self.filename_valid_degree_merge, "rb") as excel_file:
+            excel_content = excel_file.read()
+
+        expected_warnings = defaultdict(list)
+        expected_warnings[EnrollmentImporter.W_DEGREE].append(
+            'Sheet "MA Belegungen", row 3: The course\'s "Build" degree differs from it\'s degree in a previous row. Courses were automatically joined.'
+        )
+
+        success_messages, warnings, errors = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=True)
+        self.assertIn("The import run will create 1 courses/evaluations and 3 users", "".join(success_messages))
+        self.assertEqual(errors, [])
+        self.assertEqual(warnings, expected_warnings)
+
+        success_messages, warnings, errors = EnrollmentImporter.process(excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False)
+        self.assertIn("Successfully created 1 courses/evaluations, 2 students and 1 contributors", "".join(success_messages))
+        self.assertEqual(errors, [])
+        self.assertEqual(warnings, expected_warnings)
+
+        self.assertEqual(Course.objects.all().count(), 1)
+        self.assertEqual(Evaluation.objects.all().count(), 1)
+
+        course = Course.objects.get(name_de="Bauen")
+        self.assertEqual("Bachelor, Master", ", ".join(degree.name_de for degree in course.degrees.all().order_by("name_de")))
```

Comment:
compare `set(course.degrees.all())` with `set(Degrees.objects.filter(name__in=...)` or something like that? the string assembly and comparison seems weird.